tests/test_shape_inference.py

```
@@ -0,0 +1,426 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT license.
+
+"""Unit Tests for shape inference."""
+
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import numpy as np
+from onnx import TensorProto
+from tf2onnx import utils
+from tf2onnx.graph import Graph
+from backend_test_base import Tf2OnnxBackendTestBase
+from common import *  # pylint: disable=wildcard-import,unused-wildcard-import
+
+# pylint: disable=missing-docstring
+
+INPUT1 = "input1"
+INPUT2 = "input2"
+INPUT3 = "input3"
+
+
+class ShapeInferenceTests(Tf2OnnxBackendTestBase):
+    """
+    Test shape inference, it's just a subset of all cases that can be inferred shape.
+    For more information, please refer to onnx shape inference test:
+    https://github.com/onnx/onnx/blob/master/onnx/test/shape_inference_test.py
+    """
+
+    def _run_test_case(self, graph, feed_dict):
+        """Run model with onnxruntime and compare results' shape with internal shape inference."""
+        outputs = graph.outputs
+        results = []
+        raised = False
+        try:
+            results = self.run_backend(graph, outputs, feed_dict)
+        except Exception as ex:  # pylint: disable=broad-except
```

no need to try/catch, just let it fail?